### PR TITLE
Fix SMP debugging issue on rp2040

### PR DESCRIPTION
**Pull request by vishwamartur**
Related to #1172

Add `multicore_reset_core1` before `multicore_launch_core1` in `xPortStartScheduler` function to fix debugging issue on rp2040 with SMP enabled.

* Modify `portable/ThirdParty/GCC/RP2040/port.c` to include a call to `multicore_reset_core1` before `multicore_launch_core1` within the `#if portRUNNING_ON_BOTH_CORES` block.

### CODE REVIEW

File: portable/ThirdParty/GCC/RP2040/port.c [hunk 1]
```diff
@@ -22,9 +22,6 @@
  * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  *
- * https://www.FreeRTOS.org
```
Comment (ActoryOu) [inline portable/ThirdParty/GCC/RP2040/port.c:25]: Could you recover these comments back?

File: portable/ThirdParty/GCC/RP2040/port.c [hunk 2]
```diff
@@ -22,9 +22,6 @@
  * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  *
```
Comment (ActoryOu) [inline portable/ThirdParty/GCC/RP2040/port.c:24]:
```suggestion
 *
 * https://www.FreeRTOS.org
 * https://github.com/FreeRTOS
 *
```

**Review by ActoryOu — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Comment by 0x3333** [comment]
This works. For future reference: [My last comment](https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/1172#issuecomment-2451946137) in the issue, about the Timer issue this is related to.

**Comment by aggarg** [comment]
Reached out to Raspbery Pi to confirm that they are okay with the change.

**Comment by vishwamartur** [comment]
Hi @aggarg and team,

Thanks for the update and for reaching out to Raspberry Pi for confirmation. Please let me know if any additional input or verification is needed from my end. I'll be happy to assist further.

Thanks again for your support!

**Review by kilograham — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**